backend/src/models/users.go

```diff
@@ -157,4 +157,21 @@ type ResidentProgramClassInfo struct {
 	ProgramID            uint                    `json:"program_id"`
 	PresentAttendance    int                     `json:"present_attendance"`
 	AbsentAttendance     int                     `json:"absent_attendance"`
+	CreditTypes          string                  `json:"credit_types"`
+}
+
+type ResidentProgramClassHistory struct {
+	ProgramName       string      `json:"program_name"`
+	ClassName         string      `json:"class_name"`
+	Status            ClassStatus `json:"status"`
+	DateStatusChanged string      `json:"date_status_changed"`
+	CreditTypes       string      `json:"credit_types"`
+}
+
+type ResidentProgramClassWeeklySchedule struct {
+	ClassName   string `json:"class_name"`
+	StartDate   string `json:"start_date"`
+	EndDate     string `json:"end_date"`
+	CreditTypes string `json:"credit_types"`
+	Status      string `json:"status"`
 }
```

Comment:
A weekly schedule should include
- a start date and end date which represent the window that the schedule is good for (e.g., 5/1/2025 through 5/31/2025 or 5/5/2025 through 5/9/2025).
- an array of events for that schedule which include the class name, the date, start time, and duration (or end time)


I don't believe it needs to include status or credit types.